WordPress/src/main/java/org/wordpress/android/ui/jetpack/backup/BackupDownloadViewModel.kt

```diff
@@ -1,15 +1,130 @@
+import androidx.lifecycle.Transformations
 import androidx.lifecycle.ViewModel
+import kotlinx.android.parcel.Parcelize
+import org.wordpress.android.ui.jetpack.backup.BackupDownloadStep.COMPLETE
+import org.wordpress.android.ui.jetpack.backup.BackupDownloadStep.DETAILS
+import org.wordpress.android.ui.jetpack.backup.BackupDownloadStep.PROGRESS
+import org.wordpress.android.util.wizard.WizardManager
+import org.wordpress.android.util.wizard.WizardNavigationTarget
+import org.wordpress.android.util.wizard.WizardState
+import org.wordpress.android.viewmodel.SingleEventObservable
+import org.wordpress.android.viewmodel.SingleLiveEvent
 import javax.inject.Inject
 
-class BackupDownloadViewModel @Inject constructor() : ViewModel() {
-    private var isStarted: Boolean = false
-    fun start() {
-        if (isStarted) {
-            return
-        }
+const val KEY_BACKUP_DOWNLOAD_ACTIVITY_ID_KEY: String = "key_backup_download_activity_id_key"
+const val KEY_BACKUP_DOWNLOAD_CURRENT_STEP = "key_backup_download_current_step"
+const val KEY_BACKUP_DOWNLOAD_COMPLETED = "key_backup_download_completed"
+const val KEY_BACKUP_DOWNLOAD_STATE = "key_backup_download_state"
+
+@Parcelize
+@SuppressLint("ParcelCreator")
+data class BackupDownloadState(
+    val siteId: Long? = null,
+    val rewindId: Long? = null
+) : WizardState, Parcelable
+
+typealias NavigationTarget = WizardNavigationTarget<BackupDownloadStep, BackupDownloadState>
+
+class BackupDownloadViewModel @Inject constructor(
+    private val wizardManager: WizardManager<BackupDownloadStep>
+) : ViewModel() {
+    private var isStarted = false
+    private var backupDownloadCompleted = false
+
+    private lateinit var backupDownloadState: BackupDownloadState
+
+    val navigationTargetObservable: SingleEventObservable<NavigationTarget> by lazy {
+        SingleEventObservable(
+                Transformations.map(wizardManager.navigatorLiveData) {
+                    clearOldBackupDownloadState(it)
+                    WizardNavigationTarget(it, backupDownloadState)
+                }
+        )
+    }
 
+    private val _wizardFinishedObservable = SingleLiveEvent<BackupDownloadWizardState>()
+    val wizardFinishedObservable: LiveData<BackupDownloadWizardState> = _wizardFinishedObservable
+
+    private val _screenTitleObservable = SingleLiveEvent<@StringRes Int>()
+    val screenTitleObservable: LiveData<Int> = _screenTitleObservable
+
+    private val _exitFlowObservable = SingleLiveEvent<Unit>()
+    val exitFlowObservable: LiveData<Unit> = _exitFlowObservable
+
+    private val _onBackPressedObservable = SingleLiveEvent<Unit>()
+    val onBackPressedObservable: LiveData<Unit> = _onBackPressedObservable
+
+    fun start(savedInstanceState: Bundle?) {
+        if (isStarted) return
+        if (savedInstanceState == null) {
+            backupDownloadState = BackupDownloadState()
+        } else {
+            backupDownloadCompleted = savedInstanceState.getBoolean(
```

Comment:
Minor (🔍 ): Consider making all the assignment a one-liner since they fit.
```
backupDownloadCompleted = savedInstanceState.getBoolean(KEY_BACKUP_DOWNLOAD_COMPLETED, false)
backupDownloadState = requireNotNull(savedInstanceState.getParcelable(KEY_BACKUP_DOWNLOAD_STATE))
            
```

Reply:
> Consider making all the assignment a one-liner since they fit.

Running a code reformat pushes these to separate lines, so if I move them to one line and then code reformat they'll move back.

Reply:
in ef12c38 - but they'll change back, so 🤯

Reply:
If that's the case you might want to double check setup of your formatter for kotlin files - AFAIK we use 120chars per line.